ocean_lib/web3_internal/contract_base.py

```diff
@@ -199,16 +195,32 @@ def send_transaction(
         """
         contract_fn = getattr(self.contract.functions, fn_name)(*fn_args)
         contract_function = CustomContractFunction(contract_fn)
-        _transact = {
-            "from": ContractBase.to_checksum_address(from_wallet.address),
-            "account_key": from_wallet.key,
-            "chainId": self.web3.eth.chain_id,
-            "gasPrice": self.get_gas_price(self.web3),
-        }
-
-        gas_price = os.environ.get(ENV_GAS_PRICE, None)
-        if gas_price:
-            _transact["gasPrice"] = gas_price
+
+        try:
+            history = self.web3.eth.fee_history(block_count=1, newest_block="latest")
+            _transact = {
+                "from": ContractBase.to_checksum_address(from_wallet.address),
+                "account_key": from_wallet.key,
+                "chainId": self.web3.eth.chain_id,
+            }
+            _transact["gas"] = self.web3.eth.estimate_gas(_transact)
+            _transact["maxPriorityFeePerGas"] = self.web3.eth.max_priority_fee
+            _transact["maxFeePerGas"] = (
+                self.web3.eth.max_priority_fee + 2 * history["baseFeePerGas"][0]
+            )
+        except ValueError as e:
```

Comment:
will the ValueError only be encountered in fee_history? Then I suggest limiting the try to that line only. And since "account_key", "from", and "chain_id" have the same value, no matter the method chosen, please assign those first, outside the choice.

It seems to me there are in fact 3 options: when the env gas_price is set, when we are using get_gas_price and when we use the pricing history. Please make sure that all these options are clear and readable and there's a clear cause->consequence relation when reading the code. This implies better naming ("get_gas_price" is too generic now), extraction of the common keys and other small readability tweaks.

Reply:
we do not assert error messages like this, please wrap the value error only where you need it. It is generally bad practice in the code to assert an exception's text. Also, why is history passed in as a parameter? Could it be moved inside the function? Then you could treat exceptions there, and simply return None if the calc could not be calculated.

please also look into the naming suggestions I gave previously. We now have a few generic "get gas price" options, but they should reflect their meaning. One is based on history, one is based on the env, one is what? Legacy? We should clearly state this.

Reply:
`get_gas_price` is clearly for transactions that do not support EIP 1559. I wanted to include the logic of retrieving an envvar if exists for gas price calculation and also a fallback to a web3 method that evaluates the gas price for that network. I didn't see a naming suggestion for it and I think it fits well, maybe `calculate_gas_price` sounds better? 
`get_max_fee_per_gas` includes just the specific fields for EIP 1559 transactions which I think is clear what it does.
A good idea is to also provide docstrings for these functions.
I modified `get_max_fee_per_gas` the function to treat the exception inside of it and to avoid code duplication.